### PR TITLE
Initial attempt at Skipped status reasons

### PR DESCRIPTION
**Pull request by phobologic**
Fixes GH-69

### CODE REVIEW

**Comment by phobologic** [comment]
Just got a chance to test this live - this is the example output:

```
[2016-02-04T09:33:37] Plan Status:
[2016-02-04T09:33:37]   - step "mike-example-com-vpc": skipped (nochange)
[2016-02-04T09:33:37]   - step "mike-example-com-empireDB": skipped (locked)
[2016-02-04T09:33:37]   - step "mike-example-com-empireMinion": skipped (nochange)
[2016-02-04T09:33:37]   - step "mike-example-com-empireController": skipped (nochange)
[2016-02-04T09:33:37]   - step "mike-example-com-bastion": skipped (nochange)
```

File: stacker/plan.py [hunk 1]
```diff
@@ -56,7 +61,7 @@ def done(self):
     @property
     def submitted(self):
         """Returns True if the step is SUBMITTED, COMPLETE, or SKIPPED."""
-        return self.status.code >= SUBMITTED.code
+        return self.status >= SUBMITTED
```
Comment (mhahn) [inline stacker/plan.py]: fancy

File: stacker/plan.py [hunk 2]
```diff
@@ -191,7 +196,8 @@ def _single_run(self):
                 )
                 continue
 
-            if not step.done and self._watch_func and step_name not in self._watchers:
+            if not step.done and self._watch_func and \
```
Comment (mhahn) [inline stacker/plan.py]: i hate you and your weird style.

i would do this as:

```py
if (
    not step.done and
    self._watch_func and
    step_name not in self.watchers
):
```

i abhor `\` usage

File: stacker/status.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+class Status(object):
+    def __init__(self, name, code, reason=None):
+        self.name = name
+        self.code = code
+        self.reason = reason or getattr(self, 'reason', None)
+
+    # Only compares code - should it compare reason too?  I think probably
```
Comment (mhahn) [inline stacker/status.py]: i think code is fine, i would remove this comment :)

**Comment by mhahn** [comment]
+1!

**Pull request merged**